src/controllers/event.ts

```diff
@@ -39,27 +42,50 @@ export async function getPastEvents(
   next: NextFunction
 ) {
   try {
-    const eventRepository = getRepository(Event);
     const yesterday = (d => new Date(d.setDate(d.getDate() - 1)))(new Date());
-    const events = await eventRepository.find({
+    const eventsRepository = getRepository(Event);
+    const events = await eventsRepository.find({
       where: { start_datetime: LessThanOrEqual(yesterday) }
     });
-    res.status(200).send(events);
-  } catch (err) {
-    next(err);
+    prompt.successWithPayload(res, 200, "Success", events);
+  } catch (error) {
+    next(prompt.errorMessage(res, 400, error.message));
+
   }
 }
 
 export async function createEvent(
-  _req: Request,
+
+  req: Request,
+
   res: Response,
   next: NextFunction
 ) {
   try {
-    const eventRepository = getRepository(Event);
-    const newEvent = await eventRepository.save({ ..._req.body });
-    res.status(201).send(newEvent);
-  } catch (err) {
-    next(err);
+
+    const {
+      name,
+      description,
+      location,
+      start_datetime,
+      end_datetime,
+      rsvp_link,
+      media_link
+    } = req.body;
+    const event = new Event();
+    event.name = name;
+    event.description = description;
+    event.start_datetime = start_datetime;
+    event.end_datetime = end_datetime;
+    event.rsvp_link = rsvp_link;
+    event.location = location;
+    event.media_link = media_link;
+    const eventsRepository = getRepository(Event);
+    let newEvent = await eventsRepository.save(event);
```

Comment:
We can shorten this using the [spread](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) syntax to:
```
const eventsRepository = getRepository(Event);
const newEvent = eventsRepository.save({...req.body});
```

Reply:
I had considered using the spread operator which is an impressive. My question though is how shall we handle validations on individual properties within a request? Or we just rely entirely on `class-validator` package?